codegen/src/test/resources/q3/Q3.json

```diff
@@ -0,0 +1,120 @@
+{
+  "join_structure": [
+    {
+      "PrimaryRelation": "Orders",
+      "ForeignRelation": "Lineitem"
+    },
+    {
+      "PrimaryRelation": "Orders",
+      "ForeignRelation": "Customer"
+    }
+  ],
+  "LineitemProcessFunction": {
+    "name": "Q3lineitem",
+    "relation": "lineitem",
+    "this_key": [
+      "orderkey"
+    ],
+    "next_key": [
+      "orderkey"
+    ],
+    "child_nodes": 1,
+    "is_Root": true,
+    "is_Last": true,
+    "rename_attribute": null,
+    "select_conditions": {
+      "operator": "&&",
+      "value1": {
+        "left_field": {
+          "type": "attribute",
+          "name": "l_shipdate"
+        },
+        "right_field": {
+          "type": "constant",
+          "var_type": "Date",
+          "value": "1995-03-15"
+        },
+        "operator": ">"
+      }
+    }
+  },
+  "OrdersProcessFunction": {
+    "name": "Q3Orders",
+    "relation": "orders",
+    "this_key": [
+      "custkey"
+    ],
+    "next_key": [
+      "orderkey"
+    ],
+    "child_nodes": 1,
+    "is_Root": true,
+    "is_Last": true,
+    "rename_attribute": null,
+    "select_conditions": {
+      "operator": "&&",
+      "value1": {
+        "left_field": {
+          "type": "attribute",
+          "name": "orderdate"
+        },
+        "right_field": {
+          "type": "constant",
+          "var_type": "Date",
+          "value": "1995-03-15"
+        },
+        "operator": "<"
+      }
+    }
+  },
+  "CustomerProcessFunction": {
+    "name": "Q3Orders",
+    "relation": "orders",
+    "this_key": [
+      "custkey"
+    ],
+    "next_key": [
+      "custkey"
+    ],
+    "child_nodes": 1,
+    "is_Root": true,
+    "is_Last": true,
+    "rename_attribute": null,
+    "select_conditions": {
+      "operator": "&&",
+      "value1": {
+        "left_field": {
+          "type": "attribute",
+          "name": "mktsegment"
+        },
+        "right_field": {
+          "type": "constant",
+          "var_type": "varchar",
+          "value": "BUILDING"
+        },
+        "operator": "="
+      }
+    }
+  },
+  "AggregateProcessFunction": {
+    "name": "Q3Aggregate",
+    "this_key": ["orderkey"],
+    "next_key": ["orderkey","orderdate","shippriority"],
+    "delta_output": true,
+    "AggregateValue": {
```

Comment:
I think the aggregate value is defined as SUM(l_extendedprice * (1 - l_discount)) so there should be a recursive in the expression to express the (1-discount) part.

Reply:
Can you give me an example?